### ISSUE TITLE
Can not deserialize instance of MyClass out of START_ARRAY token

### ISSUE THREAD

**Issue by crumbpicker** [msgpack/msgpack-java#340]
Hi there,

I'm trying to switch from msgpack-scala to jackson-dataformat-msgpack.
I have some data already stored using msgpack-scala that I would like to deserialize using this new library.

When I try to do the deserialization I have the following Exception:
com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of MyClass out of START_ARRAY token.

What I understood is that the deserializer expect an object representation of the binary data but got an array representation. Is this right ?

Any help is welcomed.

**Comment by crumbpicker** [comment]
I found the problem.
Here is my scala class:
```
case class PeakelDataMatrix(
  val spectrumIds: Array[Long],
  val elutionTimes: Array[Float],
  val mzValues: Array[Double],
  val intensityValues: Array[Float]
)
```

When I was using msgpack-scala the serializer was encoding this class to a List of the different arrays in this class.
So now when I try to deserialize the data with the jackson module it fails because it tries to map the data inside an object (I guess a kind of bean).

I found a workaround which consists in first deserializing my data inside a Scala tuple:
```
val m = ProfiMsgPack.getObjectMapper.readValue(peaksAsBytes, classOf[(Array[Long],Array[Float],Array[Double],Array[Float])] )

val peakelDataMatrix = PeakelDataMatrix(m._1,m._2,m._3,m._4)
```

So currently I have two solutions:
1. update my stored data structure so that msgpack-jackson is happy
2. change my codebase to use this workaround

However before making a choice I would like to have some explanations:
* is there a better data representation of my data: object vs list ?
* is there a way to force jackson to deserialize the data as a List before binding the data to the class (as msgpack-scala was doing) ?

Thank you guys !

**Comment by crumbpicker** [comment]
Finally I found the answer by myself.
I have added the following annotation to my class:
```
@JsonFormat(shape=JsonFormat.Shape.ARRAY)
```

Now it serializes/deserializes correctly using the same data representation I had before.
So I see this as a bug from msgpack-scala or a behavior I don't understand from this library.

I close the issue because the origin of the problem doesn't come from jackskon-msgpack.

Sorry about that guys !
